src/algorithms/expected_consensus.md

```diff
@@ -280,35 +133,41 @@ Just as there can be 0 miners win in a round, multiple miners can be elected in
+
 Delta weight is a term composed of a few elements:
-- A wForkFactor: which seeks to cut the weight derived from rounds in which produced Tipsets do not correspond to what an honest chain is likely to have yielded (pointing to selfish mining or other non-collaborative miner behavior).
-- A wPowerFactor: which adds weight to the chain proportional to the total power backing the chain, i.e. accounted for in the chain's power table. 
-- A wBlocksFactor: which adds weight to the chain proportional to the number of blocks mined in a given round. Like wForkFactor, it rewards miner cooperation (which will yield more blocks per round on expectation).
+- wForkFactor: which seeks to cut the weight derived from rounds in which produced Tipsets do not correspond to what an honest chain is likely to have yielded (pointing to selfish mining or other non-collaborative miner behavior).
+- wPowerFactor: which adds weight to the chain proportional to the total power backing the chain, i.e. accounted for in the chain's power table. 
+- wBlocksFactor: which adds weight to the chain proportional to the number of blocks mined in a given round. Like wForkFactor, it rewards miner cooperation (which will yield more blocks per round on expectation).
 
-We have:
+The weight should be calculated using big integer arithmetic with order of operations defined above. We use brackets instead of parentheses below for legibility. We have:
 
-w[r+1] = w[r] + floor(1000 * (wForkFactor(wPowerFactor[r+1] + wBlocksFactor[r+1])))
+`w[r+1] = w[r] + (wPowerFactor[r+1] + wBlocksFactor[r+1]) * 2^8`
 
-with, for a given tipset ts in round r+1:
+For a given tipset `ts` in round `r+1`, we define:
 
-- wBlocksFactor[r+1] = v * |blocksInTipset(ts)|
-  - with v = vt * log2(totalPowerAtTipset(ts) * vs)/vs
-- wPowerFactor[r+1]  = log2(totalPowerAtTipset(ts))
-- wForkFactor[r+1]   = CDF(X, k) with X -> Bin(eNumberOfBlocksPerRound * numberOfMinersInPowerTable, 1/numberOfMinersInPowerTable)
+- `wPowerFactor[r+1]  = `
+  - with `log2b(X) = floor(log2(x)) = (binary length of X) - 1`
+    - We define the special case: `log2b(0) = 0`. Note that it should never be used (given it would mean an empty power table. 
+- wBlocksFactor[r+1] =  `wPowerFactor[r+1] * wRatio * b / e`
+  - with `b = |blocksInTipset(ts)|`
+  - `e = expected number of blocks per round in the protocol`
+  - and `wRatio in ]0, 1[`
+Thus, for stability of weight across implementations, we take: 
+- wBlocksFactor[r+1] =  `(wPowerFactor[r+1] * b * wRatio_num) / (e * wRatio_den)`
 
-The weight should be calculated using big integer arithmetic with order of operations defined above. The multiplication by 1,000 and flooring is meant to help generate uniform weights across implementations.
+We get: 
+- `w[r+1] = w[r] + log2b(totalPowerAtTipset(ts)) * 2^8 + (log2b(totalPowerAtTipset(ts)) * len(ts.blocks) * wRatio_num * 2^8) / (e * wRatio_den)`
```

Comment:
`log2b(totalPowerAtTipset(ts))` will essentially become a constant at higher network sizes (it may not catch a 2x drop in power), and it's quite painful to compute.

IMO it should be either dropped, or made more 'aggressive' (something between log and linear)

Reply:
@magik6k it is : `log2b(x big.Int) = x.BitLen() - 1`
AFAIK, it is there to prevent case where minority miner forks off and slashes other miners.

Reply:
You can still slash ~50% of power in the worst case without anyone noticing now

Reply:
Hi @magik6k, this is super useful feedback.

Remember that this is added to weight at every round, and the other factor is going to be smaller (by wRatio_num/wRatio_den), so while it is indeed a constant, a weight drop by 20% over the space of finality (say ~500 blocks) will be noticeable.

On the flip-side, a power-rental attack will have to be maintained for a bit of time before it becomes irrecoverable. Say I can rent 80% of the network for a few hours, can an honest majority collude to ignore my fork and get the heaviest back in some reasonable amount of time? This is what the more 'aggressive' functions lack...

Thereafter, @Kubuxu is right in terms of what this seeks to prevent as a main reason.

What do you think?

In any case, I need to make clear that log2b is a parameter choice as is wRatio so will update to reflect that. Log2 is simply fave candidate rn.